### ISSUE TITLE
Does not work when imported as a module.

### ISSUE THREAD

**Issue by MicahZoltu** [mozilla/webextension-polyfill#202]
If you attempt to load the `webextension-polyfill` as a module, you get a runtime failure in both Chrome and Firefox due to the utilization of `this` as the `global` parameter.  `this` is undefined in the context of an ES module, so when the root UMD module function is called and `this` is passed to the `global` parameter, it ends up passing `undefined`.  Then when `global.browser` is referenced you get a type error.

The same problem occurs with the minified source.

Error message in Firefox:
```
TypeError: global is undefined
```

### Recommendation
Use `globalThis || window || this` instead of `this` as the first parameter.  I believe any browser that supports ES modules also supports either `globalThis` or `window`, so `this` will only be hit in the rare case.

### Repro Case

```html
<!DOCTYPE html>
<html lang='en'>
	<head>
		<meta charset='utf-8'>
	</head>
	<body>
		<script type='module' src='browser-polyfill.js'></script>
	</body>
</html>
```
```json
{
	"manifest_version": 2,
	"name": "ModuleFailureRepro",
	"version": "1.0",
	"description": "Repro case for loading a polyfill as an ES module",
	"background": {
		"page": "background.html"
	}
}
```

**Comment by MicahZoltu** [comment]
A quick attempt to submit a PR shows that the problem is that the build scripts appear to be turning the script into a UMD module, and the output of that script is the problem.  Fixing this will likely require either upgrading the build script tooling, or getting the bug fixed in the build script tooling.

**Comment by MicahZoltu** [comment]
It appears the issue is coming from this buildscript step:
https://github.com/mozilla/webextension-polyfill/blob/04a9e3a2234ed61b0bd77caa7f66dc50214c9192/Gruntfile.js#L75-L80

**Comment by MicahZoltu** [comment]
I have [filed an issue](https://github.com/babel/babel/issues/10476) with Babel about this, TBD whether anything will come from that.

**Comment by rpl** [comment]
@MicahZoltu We (@Rob--W and I) discussed about this and we are open to fix this on the polyfill side by building a separate .mjs version that is going to be a pure ES6 module (and make it part of the npm package release).

**Comment by MicahZoltu** [comment]
FWIW https://github.com/babel/babel/pull/10477 has been merged.  I believe it will be available in Babel 7.7.0, whenever that release gets cut.

**Comment by rpl** [comment]
@MicahZoltu that's great (also thanks for filing it upstream).

we should definitely look into updating this repo to use a more recent version of babel (which was going to be necessary even besides this issue), I just filed #211 to keep track on it.

**Comment by rpl** [comment]
We have just landed a fix for this in #214 

I'm not closing this yet because we should also add some details about how to use the polyfill by loading as a JS module in the README.md file (and I just added a "needs: docs" label to make it visible that this is pending a pull request to do that).
